agent/engine/docker_task_engine.go

```diff
@@ -636,18 +636,59 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 		}
 	}
 
-	// Record the pullStoppedAt timestamp
-	defer func() {
-		timestamp := engine.time().Now()
-		task.SetPullStoppedAt(timestamp)
-	}()
-
-	if engine.enableConcurrentPull {
-		seelog.Infof("Task engine [%s]: pulling container %s concurrently", task.Arn, container.Name)
-		return engine.concurrentPull(task, container)
-	}
-	seelog.Infof("Task engine [%s]: pulling container %s serially", task.Arn, container.Name)
-	return engine.serialPull(task, container)
+	if engine.imagePullRequired(engine.cfg.ImagePullBehavior, container, task.Arn) {
+		// Record the pullStoppedAt timestamp
+		defer func() {
```

Comment:
In this case what's the value of task PullStartedAt and pullStoppedAt?

Reply:
I didn't quite get the question, the logic in `imagePullRequired ` is to pull image, instead of using local cache, hence recording `PullStartedAt` and `pullStoppedAt ` is necessary. (previously it's shouldUseLocalImage, and this is changed to `imagePullRequired` as per @aaithal's comments).

Reply:
So if `engine.imagePullRequired` is false, then the image pullstartedAt and pullStoppedAt won't be set, right? Can you run a task in this scenario and check the pullstoppedAt and pullstartedat from describetask API that it isn't set to something unexpected?

Reply:
yeah I see what you mean, I saw the comment here: https://github.com/aws/amazon-ecs-agent/blob/159ae5c3067e0a0d7387ff68af2a28bc90f3baaa/agent/api/task.go#L112
And I test the describe task API (which you have seen), it works as expected, the `pull started at` and `pull stopped at` will now show up.

Reply:
> the pull started at and pull stopped at will now show up.

If image is not pulled, these values shouldn't show up right?

Reply:
yeah it's not show up